Cfdi 40

AndreyPootMay opened the pull request:
Added the following catalogs in the namespace `PhpCfdi\SatCatalogos\CFDI40`:

```bash
Aduanas                 aduanas40();
ClavesUnidades          clavesUnidades40();
CodigosPostales         codigosPostales40();
FormasDePago            formasDePago40();
Impuestos               impuestos40();
MetodosDePago           metodosDePago40();
Monedas                 monedas40();
NumerosPedimentoAduana  numerosPedimentoAduana40();
Paises                  paises40();
PatentesAduanales       patentesAduanales40();
ProductosServicios      productosServicios40();
RegimenesFiscales       regimenesFiscales40();
ReglasTasaCuota         reglasTasaCuota40();
TiposComprobantes       tiposComprobantes40();
TiposFactores           tiposFactores40();
TiposRelaciones         tiposRelaciones40();
UsosCfdi                usosCfdi40();
```
Also I added the tables of sqlite3 to the seed database bash script.

All the test run successfully:

![image](https://user-images.githubusercontent.com/28898216/167475012-6196147d-e722-4915-88cc-9505bc786741.png)

Maybe we can discuss in dm all the technical stuff such as namespaces or testing and performance, greetings.

AndreyPootMay committed:
Adding cfdi40 scheme and catalogs in the library

#13

AndreyPootMay committed:
#13

eclipxe13 commented:
Es necesario seguir e implementar los esquemas de la base de datos.
Estos esquemas se pueden encontrar en <https://github.com/phpcfdi/resources-sat-catalogs/blob/master/database/schemas/>

Faltan implementar los catálogos:

- `cfdi_40_exportaciones`
- `cfdi_40_periodicidades`
- `cfdi_40_meses`
- `cfdi_40_objetos_impuestos`
- `cfdi_40_colonias`
- `cfdi_40_estados`
- `cfdi_40_localidades`
- `cfdi_40_municipios`

Los catálogos siguientes no siguen el esquema de la base de datos:

- `cfdi_40_impuestos`
- `cfdi_40_tipos_factores`
- `cfdi_40_usos_cfdi`

Te agradezco mucho el avance que has realizado y me disculpo por no haber tenido oportunidad de revisarlo antes.
**¿Te interesa completar/corregir lo que he mencionado?**

AndreyPootMay committed:
Adding new tables and updating repository

AndreyPootMay committed:
phpcs

eclipxe13 committed:
Fix cfdi_40_impuestos struct

eclipxe13 committed:
Fix cfdi_40_tipos_factores struct

eclipxe13 committed:
Fix cfdi_40_usos_cfdi

eclipxe13 committed:
Fix: remove redundant constructor

eclipxe13 committed:
Fix copy/paste tests

eclipxe13 committed:
Update seed with latest catalog database

eclipxe13 committed:
Test cfdi_40_meses on SatCatalogos

eclipxe13 committed:
Add CFDI40 Colonias (it's not an identifiable catalog)

eclipxe13 committed:
Add CFDI40_Estados (it's not an identifiable catalog)

eclipxe13 committed:
Add CFDI40_Exportaciones to repository

eclipxe13 committed:
Add CFDI40_Localidades (it's not an identifiable catalog)

eclipxe13 committed:
Add CFDI40_Municipios (it's not an identifiable catalog)

eclipxe13 committed:
Add CFDI40_Periodicidades

eclipxe13 committed:
Add CFDI40_ObjetosImpuestos

eclipxe13 committed:
Sort catalogs and tests

eclipxe13 committed:
Add test to regimenesFiscales40()

eclipxe13 committed:
Fix core maintainers group

eclipxe13 committed:
Fix typos

eclipxe13 committed:
Fix script dev:coverage

eclipxe13 committed:
Update code style & fix phpstan issues on root level

eclipxe13 committed:
Fix PHPStan issues

- Add tests to Paises
- Fix ObjetosImpuestos::create was returning Mes instead of ObjetoImpuesto
- Add internal object ScalarValues to force correct types

eclipxe13 committed:
Fix Nomina_Estado has valid dates

eclipxe13 committed:
Update build process

- Split steps to jobs
- Include PHP 8.1 on matrix build
- Update Scrutinizer to ugprade without emove and run phpunit using testdox

eclipxe13 committed:
Update license year

eclipxe13 committed:
Update "Lista de catálogos implementados"

eclipxe13 committed:
Update PHPUnit requirement and config file

eclipxe13 committed:
Review 0.3.0

- Update php-cs-fixer config file
- Add Phive
- Exclude tests/*.sql files for linguistics

eclipxe13 committed:
Update project description

eclipxe13 committed:
Prepare 0.3.0
